lib/debezium/keys_test.go

```diff
@@ -10,54 +10,85 @@ import (
 )
 
 func TestParsePartitionKeyString(t *testing.T) {
-	kv, err := parsePartitionKeyString([]byte("Struct{hi=world,foo=bar}"))
-	assert.NoError(t, err)
-	assert.Equal(t, kv["hi"], "world")
-	assert.Equal(t, kv["foo"], "bar")
-
-	kv, err = parsePartitionKeyString([]byte("Struct{hi==world}"))
-	assert.NoError(t, err)
-	assert.Equal(t, kv["hi"], "=world")
-
-	kv, err = parsePartitionKeyString([]byte("Struct{Foo=bar,abc=def}"))
-	assert.NoError(t, err)
-	assert.Equal(t, kv["foo"], "bar")
-	assert.Equal(t, kv["abc"], "def")
-
-	kv, err = parsePartitionKeyString([]byte("Struct{id=47}"))
-	assert.NoError(t, err)
-	assert.Equal(t, kv["id"], "47")
-
-	kv, err = parsePartitionKeyString([]byte("Struct{id=47,__dbz__physicalTableIdentifier=dbserver1.inventory.customers}"))
-	assert.NoError(t, err)
-	assert.Equal(t, kv["id"], "47")
-	assert.Equal(t, 1, len(kv))
-
-	kv, err = parsePartitionKeyString([]byte("Struct{uuid=d4a5bc26-9ae6-4dd4-8894-39cbcd2d526c}"))
-	assert.Nil(t, err)
-	assert.Equal(t, kv["uuid"], "d4a5bc26-9ae6-4dd4-8894-39cbcd2d526c")
-
-	badDataCases := []struct {
-		value       string
-		expectedErr string
-	}{
-		{"", "key is nil"},
-		{"Struct{", "key is too short"},
-		{"Struct{}", "key is too short"},
-		{"}", "key is too short"},
-		{"Struct{uuid=a,,}", `malformed key value pair: ""`},
-		{"Struct{,,}", `malformed key value pair: ""`},
+	{
+		// All the bad rows.
+		_, err := parsePartitionKeyString([]byte(""))
+		assert.ErrorContains(t, err, "key is nil")
+
+		_, err = parsePartitionKeyString([]byte("Struct{"))
+		assert.ErrorContains(t, err, "key is too short")
+
+		_, err = parsePartitionKeyString([]byte("Struct{}"))
+		assert.ErrorContains(t, err, "key is too short")
+
+		_, err = parsePartitionKeyString([]byte("}"))
+		assert.ErrorContains(t, err, "key is too short")
+
+		_, err = parsePartitionKeyString([]byte("Struct{uuid=a,,}"))
+		assert.ErrorContains(t, err, `malformed key value pair: ""`)
+
+		_, err = parsePartitionKeyString([]byte("Struct{,,}"))
+		assert.ErrorContains(t, err, `malformed key value pair: ""`)
 	}
+	{
+		// Valid rows
+		kv, err := parsePartitionKeyString([]byte("Struct{hi=world,foo=bar}"))
+		assert.NoError(t, err)
+		assert.Equal(t, kv["hi"], "world")
```

Comment:
The order of arguments is backwards here.

Reply:
Ah if I turn on "show whitespace" then this line isn't actually modified.

Reply:
yeah, i just copied it over. i can change the order though